examples/coap/src/oluru.rs

```diff
@@ -0,0 +1,269 @@
+//! An owned heapless container with capacity N that maintains order both through properties of its
+//! entries and by time of access -- a mix between a priority queue and an LRU cache.
+//!
+//! This was inspired by the [`uluru`](https://github.com/servo/uluru) crate. Adapting that crate
+//! to support priority levels proved to be impractical -- and many properties have changed since.
+//!
+//! See the [`OrderedPool`] documentation for details.
+//!
+//! # Terminology
+//!
+//! The ordering imposed on the cache entries is determined by a priority (see [`PriorityLevel`]).
+//! Throughout the documentation, "high" and "low" indicate priorities, whereas "small" and "large"
+//! indicate numeric values (including of priorities, where "high" corresponds to "small" and "low"
+//! to "large").
+#![deny(unsafe_code)]
```

Comment:
Thanks, applied. I'm not sure which direction this will take precisely (it started as a fork of uluru but was already completely rewritten in its image with different goals), but doing things right on the lint side is good.